synapse/replication/tcp/streams/_base.py

```diff
@@ -174,7 +170,7 @@ class BackfillStream(Stream):
     def __init__(self, hs):
         store = hs.get_datastore()
         self.current_token = store.get_current_backfill_token  # type: ignore
-        self.update_function = store.get_all_new_backfill_event_rows  # type: ignore
+        self.update_function = db_query_to_update_function(store.get_all_new_backfill_event_rows)  # type: ignore
```

Comment:
presumably if `update_function` wasn't defined as a method in the base class, the `type:ignore` wouldn't be necessary?

Reply:
Well, then it complains that `Stream` doesn't have a `update_function`, so you add one as a property and then it complains when you try and declare `async def update_function(..)` in sub classes.